tw-tasks-core/src/main/java/com/transferwise/tasks/helpers/MicrometerMeterHelper.java

```diff
@@ -24,147 +25,185 @@
+    meterRegistry.counter(METRIC_TASKS_RESUMER_STUCK_TASKS_RESUMED_COUNT, TAG_TASK_TYPE, taskType).increment();
   }
 
   @Override
   public void registerStuckTaskMarkedAsError(String taskType) {
-    meterRegistry.counter(METRIC_PREFIX + "tasksResumer.stuckTasks.markErrorCount", TAG_TASK_TYPE, taskType).increment();
+    meterRegistry.counter(METRIC_TASKS_RESUMER_STUCK_TASKS_MARK_ERROR_COUNT, TAG_TASK_TYPE, taskType).increment();
   }
 
   @Override
   public void registerStuckClientTaskResuming(String taskType) {
-    meterRegistry.counter(METRIC_PREFIX + "tasksResumer.stuckTasks.clientResumedCount", TAG_TASK_TYPE, taskType).increment();
+    meterRegistry.counter(METRIC_TASKS_RESUMER_STUCK_TASKS_CLIENT_RESUMED_COUNT, TAG_TASK_TYPE, taskType).increment();
   }
 
   @Override
   public void registerFailedTaskGrabbing(String bucketId, String taskType) {
-    meterRegistry.counter(METRIC_PREFIX + "tasks.failedGrabbingsCount", TAG_BUCKET_ID, resolveBucketId(bucketId), TAG_TASK_TYPE, taskType)
+    meterRegistry.counter(METRIC_TASKS_FAILED_GRABBINGS_COUNT, TAG_BUCKET_ID, resolveBucketId(bucketId), TAG_TASK_TYPE, taskType)
         .increment();
   }
 
   @Override
   public void registerTaskRetryOnError(String bucketId, String taskType) {
-    meterRegistry.counter(METRIC_PREFIX + "tasks.retriesCount", TAG_BUCKET_ID, resolveBucketId(bucketId), TAG_TASK_TYPE, taskType,
+    meterRegistry.counter(METRIC_TASKS_RETRIES_COUNT, TAG_BUCKET_ID, resolveBucketId(bucketId), TAG_TASK_TYPE, taskType,
         TAG_REASON, "ERROR").increment();
   }
 
   @Override
   public void registerTaskRetry(String bucketId, String taskType) {
-    meterRegistry.counter(METRIC_PREFIX + "tasks.retriesCount", TAG_BUCKET_ID, resolveBucketId(bucketId), TAG_TASK_TYPE, taskType,
+    meterRegistry.counter(METRIC_TASKS_RETRIES_COUNT, TAG_BUCKET_ID, resolveBucketId(bucketId), TAG_TASK_TYPE, taskType,
         TAG_REASON, "CONTINUE").increment();
   }
 
   @Override
   public void registerTaskResuming(String bucketId, String taskType) {
-    meterRegistry.counter(METRIC_PREFIX + "tasks.resumingsCount", TAG_BUCKET_ID, resolveBucketId(bucketId), TAG_TASK_TYPE, taskType).increment();
+    meterRegistry.counter(METRIC_TASKS_RESUMINGS_COUNT, TAG_BUCKET_ID, resolveBucketId(bucketId), TAG_TASK_TYPE, taskType).increment();
   }
 
   @Override
   public void registerTaskMarkedAsFailed(String bucketId, String taskType) {
-    meterRegistry.counter(METRIC_PREFIX + "tasks.markedAsFailedCount", TAG_BUCKET_ID, resolveBucketId(bucketId), TAG_TASK_TYPE, taskType).increment();
+    meterRegistry.counter(METRIC_TASKS_MARKED_AS_FAILED_COUNT, TAG_BUCKET_ID, resolveBucketId(bucketId), TAG_TASK_TYPE, taskType).increment();
+  }
+
+  @Override
+  public void registerTaskAdding(String type, String key, boolean inserted, ZonedDateTime runAfterTime, String data) {
+    meterRegistry.counter(METRIC_TASKS_ADDINGS_COUNT, TAG_TASK_TYPE, type, TAG_HAS_KEY, key == null ? "false" : "true",
```

Comment:
2 things. `.increment()` is missing and `x == null ? "true" : "false"` could be replaced with `Boolean.toString`, so 

```java
meterRegistry.counter(
    METRIC_TASKS_ADDINGS_COUNT,
    TAG_TASK_TYPE, type,
    TAG_HAS_KEY, Boolean.toString(key != null),
    TAG_IS_DUPLICATE, Boolean.toString(!inserted),
    TAG_IS_SCHEDULED, Boolean.toString(runAfterTime != null),
    TAG_DATA_SIZE, getDataSizeBucket(data)
).increment();
```

Reply:
thx, will fix